modules/apps/company/company-test/src/testIntegration/java/com/liferay/company/creation/performance/test/CompanyCreationPerformanceTest.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.company.creation.performance.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.AssumeTestRule;
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.util.LoggingTimer;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.util.PortalInstances;
+
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Lily Chi
+ */
+@DataGuard(scope = DataGuard.Scope.NONE)
+@RunWith(Arquillian.class)
+public class CompanyCreationPerformanceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new AssumeTestRule("assume"), new LiferayIntegrationTestRule());
+
+	public static void assume() {
+		Assume.assumeTrue(Validator.isNull(System.getenv("JENKINS_HOME")));
+	}
+
+	@Test
+	public void testSingleThreadAddCompanies() throws Exception {
+		int originalCompaniesCount = _companyLocalService.getCompaniesCount();
+
+		try (LoggingTimer loggingTimer = new LoggingTimer()) {
```

Comment:
@slnn With LoggingTimer, you don't need to manually record the time. You remove the time related logic, and move the asserting out of the `try() {}` block.

Reply:
Oh, then we can use 20794 ms in the output as the result, right?

` 2024-02-01 08:35:59.873 INFO  [com.liferay.company.test-executor-thread][LoggingTimer:35] Completed com.liferay.company.creation.performance.test.CompanyCreationPerformanceTest#testSingleThreadAddCompanies in 20794 ms`

Reply:
Done